frame/authorship/src/lib.rs

```diff
@@ -300,19 +302,19 @@ impl<T: Config> Pallet<T> {
 	///
 	/// This is safe to invoke in `on_initialize` implementations, as well
 	/// as afterwards.
-	pub fn author() -> T::AccountId {
+	pub fn author() -> Option<T::AccountId> {
```

Comment:
Trying to understand what this means... there could be no author for a block? Then who could've created it?

Reply:
> Then who could've created it?

unicorns :P 

If you read the code below you see that we have options everywhere. In normal mode there should always be an author, but in tests or other conditions it could be `None`.

Reply:
Also not sure - it might just be a gremlin in the code design.